### PR TITLE
Modify security configuration in 'app.yaml', issue #53

### PR DESCRIPTION
**Pull request by msmshariq**
In app.yaml file, modify the security config to be able to specify ignore URI's for CSRF and XSS.

Fixes issue #53

### CODE REVIEW

File: plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java [hunk 1]
```diff
@@ -356,44 +356,62 @@ public void setSubmenus(List<MenuItem> submenus) {
...
+         * Sets the list of URI's that doesn't require XSS protection.
          *
-         * @param xssPatterns XSS URI patterns to be set
+         * @param xssIgnoreUris the list of URI's that doesn't require XSS protection.
+         * @throws IllegalArgumentException if URIs in {@code xssIgnoreUris} is empty or doesn't start with '/'.
          */
-        public void setXssPatterns(PatternsConfig xssPatterns) {
-            this.xssPatterns = (xssPatterns == null) ? new PatternsConfig() : xssPatterns;
+        public void setXssIgnoreUris(List<String> xssIgnoreUris) {
+            for (String xssUri : xssIgnoreUris) {
```
Comment (this) [inline plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java]: `xssIgnoreUris` can be null (which is allowed), we need to address that as well.

File: plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java [hunk 2]
```diff
@@ -356,44 +356,62 @@ public void setSubmenus(List<MenuItem> submenus) {
...
+                if (xssUri.isEmpty()) {
```
Comment (this) [inline plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java]: Here `xssUri` can be null and we shouldn't accept such values

File: plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java [hunk 3]
```diff
@@ -356,44 +356,62 @@ public void setSubmenus(List<MenuItem> submenus) {
...
+                    throw new IllegalArgumentException("URIs configured in 'xssIgnoreUris' cannot be empty.");
```
Comment (this) [inline plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java]: Shall we change this exception message to `XSS ignore URI in the app's config cannot be empty.` (like [here](https://github.com/msmshariq/carbon-uuf-maven-tools/blob/145d73120e891b4ff3d79119ec876d63ff471458/plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java#L155))

File: plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java [hunk 4]
```diff
@@ -356,44 +356,62 @@ public void setSubmenus(List<MenuItem> submenus) {
...
+                } else if (!xssUri.startsWith("/")) {
+                    throw new IllegalArgumentException(
+                            "URIs configured in 'xssIgnoreUris' must start with a '/'. ");
```
Comment (this) [inline plugin/src/main/java/org/wso2/carbon/uuf/maven/bean/AppConfig.java]: Shall we change this exception message to `"XSS ignore URI in the app's config must start with a '/'. Instead found '" +xssUri.charAt(0) + "' at the beginning of the ignore URI '" + xssUri + "'."`

File: plugin/src/test/java/org/wso2/carbon/uuf/maven/bean/AppConfigTest.java [hunk 5]
```diff
@@ -105,18 +105,18 @@ public void testMenuValidations() {
...
-        patternsConfig.setReject(null);
-        patternsConfig.setReject(Collections.emptyList());
-        patternsConfig.setReject(Collections.singletonList("/some/pattern"));
+                () -> securityConfig.setCsrfIgnoreUris(Collections.singletonList("some/uri")));
+        Assert.assertThrows(IllegalArgumentException.class,
+                () -> securityConfig.setXssIgnoreUris(Collections.singletonList("")));
+        Assert.assertThrows(IllegalArgumentException.class,
+                () -> securityConfig.setXssIgnoreUris(Collections.singletonList("some/uri")));
+
+        securityConfig.setCsrfIgnoreUris(Collections.singletonList("/valid/uri"));
```
Comment (this) [inline plugin/src/test/java/org/wso2/carbon/uuf/maven/bean/AppConfigTest.java]: "CSRF ignore URI"s and "XSS ignore URI"s can be `null`, and its acceptable. We need to test it here.

**Comment by msmshariq** [comment]
Implemented the suggested improvements.

**Pull request merged**